exporters/zipkin/src/test/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporterEndToEndHttpTest.java

```diff
@@ -0,0 +1,195 @@
+            .setEncoder(SpanBytesEncoder.THRIFT)
+            .build();
+
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportAsJsonV1() {
+
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
+            .setSender(
+                URLConnectionSender.newBuilder()
+                    .endpoint(zipkin.httpUrl() + ENDPOINT_V1_SPANS)
+                    .encoding(Encoding.JSON)
+                    .build())
+            .setServiceName(SERVICE_NAME)
+            .setEncoder(SpanBytesEncoder.JSON_V1)
+            .build();
+
+    exportAndVerify(configuration);
+  }
+
+  @Test
+  public void testExportFailedAsWrongEncoderUsed() {
+
+    ZipkinExporterConfiguration configuration =
+        ZipkinExporterConfiguration.builder()
+            .setSender(
+                URLConnectionSender.newBuilder()
+                    .endpoint(zipkin.httpUrl() + ENDPOINT_V2_SPANS)
+                    .encoding(Encoding.JSON)
+                    .build())
+            .setServiceName(SERVICE_NAME)
+            .setEncoder(SpanBytesEncoder.PROTO3)
+            .build();
+
+    ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.create(configuration);
+
+    SpanData spanData = buildStandardSpan().build();
+    SpanExporter.ResultCode resultCode = zipkinSpanExporter.export(Collections.singleton(spanData));
+
+    assertThat(resultCode).isEqualTo(SpanExporter.ResultCode.FAILURE);
+    List<zipkin2.Span> zipkinSpans = zipkin.getTrace(TRACE_ID);
+    assertThat(zipkinSpans).isNotNull();
+    assertThat(zipkinSpans).isEmpty();
+  }
+
+  /**
+   * Exports a span, verify that it was received by Zipkin, and check that the span stored by Zipkin
+   * matches what was sent.
+   */
+  private void exportAndVerify(ZipkinExporterConfiguration configuration) {
+    ZipkinSpanExporter zipkinSpanExporter = ZipkinSpanExporter.create(configuration);
+
+    SpanData spanData = buildStandardSpan().build();
+    SpanExporter.ResultCode resultCode = zipkinSpanExporter.export(Collections.singleton(spanData));
+
+    assertThat(resultCode).isEqualTo(SpanExporter.ResultCode.SUCCESS);
+    List<zipkin2.Span> zipkinSpans = zipkin.getTrace(TRACE_ID);
+
+    assertThat(zipkinSpans).isNotNull();
+    assertThat(zipkinSpans.size()).isEqualTo(1);
+    assertThat(zipkinSpans.get(0))
+        .isEqualTo(ZipkinSpanExporter.generateSpan(spanData, localEndpoint));
+  }
+
+  private static SpanData.Builder buildStandardSpan() {
+    return SpanData.newBuilder()
+        .setTraceId(TraceId.fromLowerBase16(TRACE_ID, 0))
+        .setSpanId(SpanId.fromLowerBase16(SPAN_ID, 0))
+        .setParentSpanId(SpanId.fromLowerBase16(PARENT_SPAN_ID, 0))
+        .setTraceFlags(TraceFlags.builder().setIsSampled(true).build())
+        .setStatus(Status.OK)
+        .setKind(Span.Kind.SERVER)
+        .setHasRemoteParent(true)
+        .setName("Recv.helloworld.Greeter.SayHello")
+        .setStartEpochNanos(1505855794_194009601L)
```

Comment:
interesting using nanos resolution, but will leave that aside..

when you add a method that generates the zipkin span, probably moving these timestamps to constants will help, and show that in zipkin they are /1000